front/lib/api/assistant/conversation.ts

```diff
@@ -630,16 +570,167 @@ export async function* retryAgentMessage(
   | AgentGenerationSuccessEvent
   | AgentMessageSuccessEvent
 > {
-  yield {
-    type: "agent_error",
-    created: Date.now(),
-    configurationId: "foo",
-    messageId: "bar",
-    error: {
-      code: "not_implemented",
-      message: "Not implemented",
+  const {
+    event,
+    agentMessageRow,
+  }: {
+    event: AgentMessageNewEvent | AgentErrorEvent;
+    agentMessageRow?: AgentMessage;
+  } = await front_sequelize.transaction(async (t) => {
+    const messageRow = await Message.findOne({
+      where: {
+        conversationId: conversation.id,
+        id: message.id,
+      },
+      include: [
+        {
+          model: AgentMessage,
+          as: "agentMessage",
+          required: true,
+        },
+      ],
+      transaction: t,
+    });
+
+    if (!messageRow) {
```

Comment:
if (!messageRow || !messageRow.agentMessage)

and remove the other branch below